checker/src/main/java/org/checkerframework/checker/resourceleak/ResourceLeakVisitor.java

```diff
@@ -366,6 +300,43 @@ public Void visitVariable(VariableTree tree, Void p) {
     return super.visitVariable(tree, p);
   }
 
+  /**
+   * An obligation that must be satisfied by a destructor. Helper type for {@link
+   * #checkOwningField(Element)}.
+   */
+  private static final class DestructorObligation {
```

Comment:
If we're keeping note of such things, this could also be a record eventually.  I'm ok with not adding such comments everywhere.